components/constellation/constellation.rs

```diff
@@ -2806,23 +2805,25 @@ where
             ExitPipelineMode::Force,
         );
 
-        let failure_url = ServoUrl::parse("about:failure").expect("infallible");
-
-        if pipeline_url == failure_url {
-            return error!("about:failure failed");
+        let old_pipeline_id = pipeline_id;
+        let old_load_data = match self.pipelines.get(&pipeline_id) {
+            Some(pipeline) => pipeline.load_data.clone(),
+            None => return warn!("failed pipeline is missing"),
+        };
+        if old_load_data.crash.is_some() {
+            return error!("crash page crashed");
         }
 
-        warn!("creating replacement pipeline for about:failure");
+        warn!("creating replacement pipeline for crash page");
 
         let new_pipeline_id = PipelineId::new();
-        let load_data = LoadData::new(
-            LoadOrigin::Constellation,
-            failure_url,
-            None,
-            Referrer::NoReferrer,
-            None,
-            None,
-        );
+        let mut new_load_data = old_load_data.clone();
+        new_load_data.crash = Some(if let Some(backtrace) = backtrace {
+            format!("{}\n{}", reason, backtrace)
+        } else {
+            reason
+        });
+
```

Comment:
I think this could be slightly more idiomatically:

```rust
new_load_data.crash = backtrace.map(|backtrace|
    format!("{}\n{}", reason, backtrace)
).unwrap_or(reason);
```

Reply:
done in db246afdd505aef117a00a9060d30846d9705fff